src/dev/build/tasks/copy_source_task.ts

```diff
@@ -60,5 +65,55 @@ export const CopySource: Task = {
         'opensearch_dashboards.d.ts',
       ],
     });
+
+    /*
+     * Dependencies linked using `file:` in the `yarn.lock` file are
+     * referenced with relative paths. Moving them to the `build`
+     * folder, these paths need to be transformed.
+     */
+    const writeStream = createWriteStream(join(buildRoot, 'yarn.lock'));
+    const writeLine = (line: string) =>
+      new Promise((resolve) => {
+        if (writeStream.write(line + '\n')) return resolve();
+        writeStream.once('drain', resolve);
+      });
+
+    /*
+     * Dependency patterns are comma separated `name@range` values, followed
+     * by a colon, like `AAAA:` and `AAAA, BBBB, CCCC:`
+     * Some are wrapped in quotes, like `"AAAA":` and `"AAAA, BBBB, CCCC":`
+     *
+     * The linked file will have `@file:` in it and the inner strings will
+     * not have commas or colons.
+     *
+     * Zero-width lookbehind and lookaheads allow getting only the part
+     * required for processing the patterns. `\S` is to avoid matching
+     * spaces after commas.
+     */
+    const linkedFileDepPattern = /(?<=^"?|,\s*)(\S+?[^,:]+)@file:([^,:]+)(?=,\s*|"?:\s*$)/g;
+    const fileLinkDelimiter = '@file:';
+
+    const reader = createInterface({
+      input: createReadStream(join(repoRoot, 'yarn.lock')),
+      crlfDelay: Infinity,
+    });
+
+    for await (const line of reader) {
+      /*
+       * For added safety, we make sure the line doesn't start with whitespace
+       * before we check that it has `@file:`.
+       */
```

Comment:
So, the intended outcome here is that the moved `yarn.lock` file should not contain any `@file:` URIs, right? It seems like we could also do a final check of the updated `yarn.lock` to make sure that's the case and throw an error if there are some that weren't correctly resolved. Not sure it's necessary, but it would help us immediately know that something in this task isn't working as intended.

Reply:
> So, the intended outcome here is that the moved yarn.lock file should not contain any @file: URIs, right?

It will have `file:` URIs that point to the correct place. We don't test anything during a build so I am torn between adding a single line to check if the path exists or continuing status quo.

Reply:
Ah, got it - we'll still have `file:` URIs, they'll just be absolute, not relative.